test/core/nexus/NexusRemoteTests.cpp

```diff
@@ -25,79 +25,78 @@ class Nexus_Remote_Test : public ::testing::Test {
 
     void TearDown() override;
 
-    std::shared_ptr<HY_PointHydroNexusRemoteUpstream> upstream_remote_nexus;
-    std::shared_ptr<HY_PointHydroNexusRemote> downstream_remote_nexus;
+    std::vector<double> stored_discharge;
+    int mpi_rank;
 };
 
-void Nexus_Remote_Test::SetUp() {
-   int nexus_id_number = 26;
-   std::string nexus_id = "nex-26";
-   int num_downstream = 1;
-
-   MPI_Init(NULL, NULL);
-
-   upstream_remote_nexus = std::make_shared<HY_PointHydroNexusRemoteUpstream>(nexus_id_number, nexus_id, num_downstream);
+void Nexus_Remote_Test::SetUp()
+{
+    MPI_Init(NULL, NULL);
+    MPI_Comm_rank(MPI_COMM_WORLD, &mpi_rank);
 
-   std::unordered_map<long,long> loc_map;
-   loc_map[27] = 0;
-   downstream_remote_nexus = std::make_shared<HY_PointHydroNexusRemote>(nexus_id_number, nexus_id, num_downstream, loc_map);
+    stored_discharge.push_back(1.2);
+    stored_discharge.push_back(1.8);
+    stored_discharge.push_back(2.3);
+}
 
+void Nexus_Remote_Test::TearDown()
+{
+    MPI_Finalize();
 }
 
-void Nexus_Remote_Test::TearDown() {
 
-}
 
 //Test sending data with MPI from an upstream remote nexus
 //to a downstream remote nexus.
 TEST_F(Nexus_Remote_Test, TestInit0)
 {
-   int send_catchment_id = 27;
-   double send_flow = 1.2;
-   long send_time_step = 3;
-
-   int receive_catchment_id = 0;
-   double receive_flow = 0.0;
-   long receive_time_step = 0;
-
-   // Find out rank, size
-   int world_rank;
-   MPI_Comm_rank(MPI_COMM_WORLD, &world_rank);
+    HY_PointHydroNexusRemote::catcment_location_map_t loc_map;
 
-   // If the world_rank is 0, then call HY_PointHydroNexusRemoteUpstream to send
-   // data to downstream remote nexus.
-   if (world_rank == 0)
-   {
-      upstream_remote_nexus->add_upstream_flow(send_flow, send_catchment_id, send_time_step);
-   }
+    HY_PointHydroNexusRemote* nexus;
 
-   // Else if the world_rank is 1, then call HY_PointHydroNexusRemoteDownstream to receive data
-   // from the upstream remote nexus.
-   else if (world_rank == 1)
-   {
-      downstream_remote_nexus->add_upstream_flow(-9999.0, send_catchment_id, send_time_step);
-
-      receive_catchment_id = 27;
-      receive_flow = downstream_remote_nexus->inspect_upstream_flows(send_time_step).first;
-      receive_time_step = downstream_remote_nexus->get_time_step();
-
-      std::cout << "Received: \nCatchment ID " << receive_catchment_id << " \nFlow "
-      << receive_flow << " \nTime Step " << receive_time_step << std::endl;
-
-      //Assert equal for the send and receive values
-      EXPECT_DOUBLE_EQ(send_flow, receive_flow);
+    // create a nexus at both ranks
+    if ( mpi_rank == 0)
+    {
+        loc_map[26] = 1;
+        nexus = new HY_PointHydroNexusRemote(27, "nexus-27", 1, loc_map);
```

Comment:
Did you try this again with a smart pointer?

Reply:
The test is changed to use smart pointers for the nexus objects.